src/core/positioning/internalgnssreceiver.cpp

```diff
@@ -103,20 +116,81 @@ void InternalGnssReceiver::handlePositionUpdated( const QGeoPositionInfo &positi
                                                             elevation,
                                                             speed,
                                                             direction,
-                                                            QList<QgsSatelliteInfo>(), 0, 0, 0,
+                                                            mSatellitesInfo, 0, 0, 0,
                                                             hacc,
                                                             vacc,
                                                             positionInfo.timestamp(),
-                                                            QChar(), 0, -1, 0, QChar( 'A' ), QList<int>(), false,
+                                                            QChar(), 0, -1, mSatellitesUsed, QChar( 'A' ), mSatellitesID, mSatelliteInformationValid,
                                                             verticalSpeed,
                                                             magneticVariation,
                                                             0, mGeoPositionSource->sourceName() );
     emit lastGnssPositionInformationChanged( mLastGnssPositionInformation );
   }
 }
 
+void InternalGnssReceiver::handleSatellitesInUseUpdated( const QList<QGeoSatelliteInfo> &satellites )
+{
+  const int satellitesUsed = satellites.size();
```

Comment:
According to the docs https://doc.qt.io/qt-5/qgeosatelliteinfosource.html#Error-enum

> If satellite information cannot be retrieved or some other form of timeout has occurred the [satellitesInViewUpdated](https://doc.qt.io/qt-5/qgeosatelliteinfosource.html#satellitesInViewUpdated)() and [satellitesInUseUpdated](https://doc.qt.io/qt-5/qgeosatelliteinfosource.html#satellitesInUseUpdated)() signals may be emitted with empty parameter lists.

we should reset all values if `satellites.isEmpty()`

Reply:
Hm, good question. I'm wondering whether we should actually just skip here, to be tolerant (I'm not sure if we need to be here but for positioning we have to as info can go missing for a given reading)